.github/workflows/tests_linux_x86_mpi.yml

```diff
@@ -252,15 +248,75 @@ jobs:
       - name: Run PennyLane-Lightning-GPU unit tests
         run: |
           source /etc/profile.d/modules.sh && module use /opt/modules/ && module load ${{ matrix.mpilib }}
-          PL_DEVICE=lightning.gpu python -m pytest ./tests/
-          PL_DEVICE=lightning.gpu /opt/mpi/${{ matrix.mpilib }}/bin/mpirun -np 2 python -m pytest ./mpitests/test_adjoint_jacobian.py
+          PL_DEVICE=lightning.gpu python -m pytest ./tests/ $COVERAGE_FLAGS
+          PL_DEVICE=lightning.gpu /opt/mpi/${{ matrix.mpilib }}/bin/mpirun -np 2 python -m pytest ./mpitests $COVERAGE_FLAGS
+          # PL_DEVICE=lightning.gpu /opt/mpi/${{ matrix.mpilib }}/bin/mpirun --oversubscribe -n 4 pytest -s -x mpitests/test_device.py -k test_create_device
+          mv coverage.xml coverage-${{ github.job }}-lightning_gpu_mpi.xml
+
+      - name: Upload code coverage results
+        uses: actions/upload-artifact@v3
+        with:
+          name: ubuntu-codecov-results-python
+          path: coverage-${{ github.job }}-lightning_gpu_mpi.xml
+          if-no-files-found: error
+
+      - name: Cleanup
+        if: always()
+        run: |
+          rm -rf ${{ steps.setup_venv.outputs.venv_name }}
+          rm -rf * .git .gitignore .github
+          pip cache purge
+
+  upload-to-codecov-linux-cpp:
```

Comment:
will the results of `openmpi` test overrid the `mpich` ones?

Reply:
Good question, maybe the last one is the one to report? Is that important?

Reply:
yea, `codecov` would complain that there is lack of coverage for `mpich` or `openmpi`.